examples/jsm/renderers/webgl/nodes/WebGLNodeBuilder.js

```diff
@@ -460,29 +466,45 @@ class WebGLNodeBuilder extends NodeBuilder {
 
 		const uniforms = this.uniforms[ shaderStage ];
 
-		let snippet = '';
+		let output = '';
 
 		for ( const uniform of uniforms ) {
 
+			let snippet = null;
+
 			if ( uniform.type === 'texture' ) {
 
-				snippet += `uniform sampler2D ${uniform.name}; `;
+				snippet = `sampler2D ${uniform.name}; `;
 
 			} else if ( uniform.type === 'cubeTexture' ) {
 
-				snippet += `uniform samplerCube ${uniform.name}; `;
+				snippet = `samplerCube ${uniform.name}; `;
 
 			} else {
 
 				const vectorType = this.getVectorType( uniform.type );
 
-				snippet += `uniform ${vectorType} ${uniform.name}; `;
+				snippet = `${vectorType} ${uniform.name}; `;
+
+			}
+
+			const precision = uniform.node.precision;
+
+			if ( precision !== 'medium' ) {
+
+				snippet = 'uniform ' + precisionLib[ precision ] + ' ' + snippet;
```

Comment:
I think the default precision in GLSL isn't always mediump, it's more complex...

Reply:
I think we are working with relative values, would be the estimated accuracy of the hardware. low (minimal), medium( is default), high( max)

Reply:
> Maybe we should move it to UniformNode? Because I don't think precisions are useful for ConstNodes.

Hmm... I think it might be useful to store a variable that requires high precision like PI?

Reply:
I mean that something like `uniform mediump sampler2D name` and `uniform sampler2D name` aren't equivalent -- the latter has lowp precision. So it's better to always specify precision (i.e. remove the check of precision for mediump).

Reply:
> Hmm... I think it might be useful to store a variable that requires high precision like PI?

I think our ConstNodes are always inlined -- and inline literals don't need to specify precision in GLSL (and I don't think it's possible?).

Reply:
> I think our ConstNodes are always inlined -- and inline literals don't need to specify precision in GLSL (and I don't think it's possible?).

I think we can have variables that will store calculations of constants or itself if one or more constants have the precision flag, in the same way that we preserve the greatest length when we add `float() + vec3()`, `lowp + highp` should return `highp`.

Reply:
OK, agree then.